File: app/src/main/java/com/tentlers/mngapp/data/tables/bills/BillItemForCard.java
package com.tentlers.mngapp.data.tables.bills;


import java.util.Date;

import androidx.room.ColumnInfo;
import androidx.room.Ignore;

/* this is used to handle all the date used in the each card of bill.*/
public class BillItemForCard {

    @ColumnInfo(name = "billId")
    public long billId;

    @ColumnInfo(name = "createDate")
    public Date createDate;

    @ColumnInfo(name = "isBillPaid")
    public boolean isBillPaid;

    @ColumnInfo(name = "electricCost")
    public float electricCost;
    /*TODO: change electric cost to electric charges.*/

    @ColumnInfo(name = "monthlycharge")
    public float monthlycharge;

    @ColumnInfo(name = "additionalcharge")
    public float additionalcharge;

    @ColumnInfo(name = "totalAmt")
    public float totalAmt;

    @ColumnInfo(name = "tenantId")
    public long tenantId;

    @ColumnInfo(name = "tenantName")
    public String tenantName;

    @ColumnInfo(name = "houseName")
    public String houseName;

    @ColumnInfo(name = "roomName")
    public String roomName;

    public static final boolean VIEW_MORE = true;
    public static final boolean VIEW_LESS = false;
    @Ignore
    public boolean amtDescState;


}

File: app/src/main/java/com/tentlers/mngapp/data/tables/bills/BillItemForCard.kt
package com.tentlers.mngapp.data.tables.bills

import androidx.room.ColumnInfo
import androidx.room.Ignore
import java.util.Date

/* this is used to handle all the date used in the each card of bill.*/
class BillItemForCard {
    @ColumnInfo(name = "billId")
    var billId: Long = 0

    @ColumnInfo(name = "createDate")
    var createDate: Date? = null

    @ColumnInfo(name = "isBillPaid")
    var isBillPaid = false

    @ColumnInfo(name = "electricCost")
    var electricCost = 0f

    /*TODO: change electric cost to electric charges.*/
    @ColumnInfo(name = "monthlyCharge")
    var monthlyCharge = 0f

    @ColumnInfo(name = "additionalCharge")
    var additionalCharge = 0f

    @ColumnInfo(name = "totalAmt")
    var totalAmt = 0f

    @ColumnInfo(name = "tenantId")
    var tenantId: Long = 0

    @ColumnInfo(name = "tenantName")
    var tenantName: String? = null

    @ColumnInfo(name = "houseName")
    var houseName: String? = null

    @ColumnInfo(name = "roomName")
    var roomName: String? = null

    @JvmField
    @Ignore
    var amtDescState = false

    companion object {
        const val VIEW_MORE = true
        const val VIEW_LESS = false
    }
}